stake-pool/single-pool/src/processor.rs

```diff
@@ -621,18 +722,26 @@ impl Processor {
         let token_program_info = next_account_info(account_info_iter)?;
         let stake_program_info = next_account_info(account_info_iter)?;
 
-        check_pool_stake_address(program_id, vote_account_address, pool_stake_info.key)?;
-        let bump_seed = check_pool_authority_address(
+        let pool = SinglePool::from_account_info(program_id, pool_info)?;
+        check_pool_address(program_id, &pool.vote_account_address, pool_info.key)?;
```

Comment:
oh, maybe i should just delete it because it sends the wrong message. its actually completely pointless, i just thought it was cute for completeness

the vote address is set in `process_initialize`, where its confirmed that the pool account address properly derives from the vote account. so this really is more of a "ghosts didnt alter the data" thing. the important checks are all in `SinglePool::from_account_info` where its checked that the pool account is owned by the program and has the first byte incremented

incidentally while rechecking the code i realize i should move `check_account owner` into `lib.rs` and use that in `SinglePool::from_account_info` instead of the if statement so the errors are consistent

Reply:
Gotcha, makes sense to me -- it's up to you! the auditors might end up saying "this address is unchecked" anyway :upside_down_face: